packages/expo/cli/utils/profile.ts

```diff
@@ -0,0 +1,39 @@
+import chalk from 'chalk';
+
+import * as Log from '../log';
+
+/**
+ * Wrap a method and profile the time it takes to execute the method using `EXPO_PROFILE`.
+ * Works best with named functions (i.e. not arrow functions).
+ *
+ * @param fn function to profile.
+ * @param functionName optional name of the function to display in the profile output.
+ */
+export const profile = <T extends any[], U>(
+  fn: (...args: T) => U,
+  functionName: string = fn.name
+): ((...args: T) => U) => {
+  const name = chalk.dim(`⏱  [profile] ${functionName ?? 'unknown'}`);
+  return (...args: T): U => {
+    Log.time(name);
+    const results = fn(...args);
+    if (results instanceof Promise) {
+      // @ts-ignore: Type 'Promise<U>' is not assignable to type 'U'. 'U' could be instantiated with an arbitrary type which could be unrelated to 'Promise<U>'.
+      return new Promise<U>((resolve, reject) => {
```

Comment:
The error makes sense.. this should be `Promise<Awaited<U>>`. Still will get a type check error since it's unsound but at least the type declaration will be right